src/components/progress-circle/progress-circle.ts

```diff
@@ -41,47 +84,172 @@ export class MdProgressCircle {
+        // if the animation has lasted long than the animation duration.
+        if (id === this._lastAnimationId && elapsedTime < animationDuration) {
+          window.requestAnimationFrame(animation);
+        }
+      };
+      window.requestAnimationFrame(animation);
+    }
   }
 
 
-  /** Gets the progress value, returning the clamped value. */
-  get value() {
-    return this._value;
+  /**
+   * Starts the indeterminate animation interval, if it is not already running.
+   */
+  private _startIndeterminateAnimation() {
+    var rotationStartPoint = 0;
+    var start = startIndeterminate;
+    var end =  endIndeterminate;
+    var duration = durationIndeterminate;
+    var animate = () => {
+      this._renderCircle(start, end, MdProgressCircle.materialEase, duration, rotationStartPoint);
+      // Prevent rotation from reaching Number.MAX_SAFE_INTEGER.
+      rotationStartPoint = (rotationStartPoint + end) % 100;
+      let temp = start;
+      start = -end;
+      end = -temp;
+    };
+
+    if (!this._interdeterminateInterval) {
+      // Note that this interval isn't supposed to trigger a digest.
+      this._interdeterminateInterval = window.setInterval(
+        animate, durationIndeterminate + 50, 0, false);
+      animate();
+    }
   }
 
 
-  /** Sets the progress value, clamping before setting the internal value. */
-  set value(v: number) {
-    if (v != null) {
-      this._value = MdProgressCircle.clamp(v);
+  /**
+   * Removes interval, ending the animation.
+   */
+  private _cleanupIndeterminateAnimation() {
+    if (this._interdeterminateInterval) {
+      window.clearInterval(this._interdeterminateInterval);
+      this._interdeterminateInterval = null;
     }
   }
 
 
+  /**
+   * Determines the path value to define the arc.
+   */
+  private static getSvgArc(current: number, rotation: number) {
+    // The angle can't be exactly 360, because the arc becomes hidden.
+    var maximumAngle = 359.99 / 100;
+    var startPoint = rotation || 0;
+    var radius = 50;
+    var pathRadius = 40;
+
+    var startAngle = startPoint * maximumAngle;
+    var endAngle = current * maximumAngle;
+    var start = MdProgressCircle.polarToCartesian(radius, pathRadius, startAngle);
+    var end = MdProgressCircle.polarToCartesian(radius, pathRadius, endAngle + startAngle);
+    var arcSweep = endAngle < 0 ? 0 : 1;
+    var largeArcFlag: number;
+
+    if (endAngle < 0) {
+      largeArcFlag = endAngle >= -180 ? 0 : 1;
+    } else {
+      largeArcFlag = endAngle <= 180 ? 0 : 1;
+    }
+
+    return `M${start}A${pathRadius},${pathRadius} 0 ${largeArcFlag},${arcSweep} ${end}`;
+  }
+
+
   /** Clamps a value to be between 0 and 100. */
-  static clamp(v: number) {
+  private static clamp(v: number) {
     return Math.max(0, Math.min(100, v));
   }
-}
 
 
+  /**
+   * Easing function for linear animation.
+   */
+  private static linearEase(t: number, b: number, c: number, d: number) {
```

Comment:
This looks like it was modeled after one of the Robert Penner easing functions:
`// t: current time, b: begInnIng value, c: change In value, d: duration`

Maybe expand these names to make it more obvious what is going on.